roles/kube_proxy_and_dns/tasks/main.yml

```diff
@@ -1,5 +1,6 @@
 ---
 - name: Ensure project exists
+  delegate_to: "{{ groups.oo_first_master.0 }}"
```

Comment:
This role is called via import_role from another role (calico).  It seems that run_once is not being applied as it should (or ansible does not support run_once on import_role?)

We should put this entire task file in a block, use run_once here, and when ansible_hostname == groups.oo_first_master.0

Reply:
Do you mean this entire main.yaml should be changed to something like the following?
~~~
- when: ansible_hostname == groups.oo_first_master.0
  block:
    - ensure project exists
      run_once: true
[ ... ]
    - all tasks here
~~~

Or can we just simply import this role with from calico with:
---
- block:
  - name: Calico | Run kube proxy
    import_role:
        name: kube_proxy_and_dns
  run_once: True
~~~

OR 

~~~
- name: Calico | Run kube proxy
  import_role:
    name: kube_proxy_and_dns
  when: 'ansible_hostname == groups.oo_first_master.0
~~~

Reply:
On second though, let's just apply the when statement I suggested onto the import_role in calico, that should pretty much cover what we need.  I'm unsure how block might interpret run_once, but I know how import_role + when works.

Remove run_once statement from import_role in calico.

Reply:
Also, add a comment explaining why we're doing that there so we don't regress on it in the future.

Reply:
Done, delegate_to deleted, changed run_once for when and added a comment documenting the change.